crates/dbsp/src/trace/layers/persistent/ordered_layer/mod.rs

```diff
@@ -0,0 +1,1416 @@
+//! Implementation using ordered keys and exponential search.
+
+//pub use consumer::{OrderedLayerConsumer, OrderedLayerValues};
+use std::{
+    cmp::{min, Ordering},
+    fmt::{Debug, Display, Formatter},
+    ops::{Add, AddAssign, Neg},
+};
+
+use crate::trace::Deserializable;
+use crate::{
+    algebra::{AddAssignByRef, AddByRef, NegByRef},
+    trace::layers::persistent::column_layer::cursor::ScrapSpace,
+    trace::layers::{
+        advance, retreat, Builder, Cursor, MergeBuilder, OrdOffset, Trie, TupleBuilder,
+    },
+    DBData, NumEntries,
+};
+use rand::Rng;
+use rkyv::{AlignedVec, Archive, Deserialize, Infallible, Serialize};
+use size_of::SizeOf;
+use textwrap::indent;
+
+use super::utils::sample_slice;
+use super::Persistence;
+
+use feldera_storage::backend::metadata::{FileHeader, Metadata};
+
+/// A level of the trie, with keys and offsets into a lower layer.
+///
+/// In this representation, the values for `keys[i]` are found at `vals[offs[i]
+/// .. offs[i+1]]`.
+// False positive from clippy
```

Comment:
what does this comment refer to?

Reply:
I think that everything under `persistent` is not going to be committed.